packages/kbn-unified-data-table/src/components/data_table_document_selection.tsx

```diff
@@ -119,12 +134,48 @@ export function DataTableDocumentToolbarBtn({
             setIsFilterActive(true);
           }}
         >
-          <FormattedMessage
-            id="unifiedDataTable.showSelectedDocumentsOnly"
-            defaultMessage="Show selected documents only"
-          />
+          {isPlainRecord ? (
+            <FormattedMessage
+              id="unifiedDataTable.showSelectedResultsOnly"
+              defaultMessage="Show selected results only"
+            />
+          ) : (
+            <FormattedMessage
+              id="unifiedDataTable.showSelectedDocumentsOnly"
+              defaultMessage="Show selected documents only"
+            />
+          )}
         </EuiContextMenuItem>
       ),
+    ];
+
+    if (enableComparisonMode && selectedDocs.length > 1) {
```

Comment:
I'm with Julia and I actually wanted to go a step further and ask that we consider showing something like this when more than 1 document has been selected. It will give this feature the discoverability it deserves.
<img width="837" alt="image" src="https://github.com/elastic/kibana/assets/4016496/443b902b-68d3-415c-a77f-b60b659480f8">